src/main/java/use_cases/create_new_month/NewMonthUCI.java

```diff
@@ -0,0 +1,61 @@
+package use_cases.create_new_month;
+
+import entities.*;
+
+import java.util.ArrayList;
+
+/**
+ * The use case interactor class for creating a new MonthlyStorage.
+ * This class implements the NewMonthIB interface. The controller
+ * class calls this class to get the NewMonthOD object.
+ */
+public class NewMonthUCI implements NewMonthIB {
+    final NewMonthOB outputBoundary;
+
+    /**
+     * Construct a NewMonthUCI.
+     * @param outputBoundary NewMonthOB related to using output
+     */
+    public NewMonthUCI(NewMonthOB outputBoundary) {
+        this.outputBoundary = outputBoundary;
+    }
+
+    /**
+     * Pass in and use NewMonthID containing input data to create output data.
+     * @param input input passed in from the controller class
+     * @return NewMonthOD object that contains output data
+     */
+    @Override
+    public NewMonthOD createOutput(NewMonthID input) {
+        SessionStorage session = input.getSession();
+        int monthID = input.getMonthID();
+        double budgetValue = input.getBudgetValue();
+
+        // Check that no MonthlyStorage in this session has the same monthID
+        ArrayList<MonthlyStorage> allMonthArrayList = session.getAllMonthlyData();
```

Comment:
I think this could be put into a private method to make the code shorter. That, or it could be implemented into SessionStorage as a method there since that would make sense as well.